src/platform/qt/library/LibraryModel.cpp

```diff
@@ -0,0 +1,424 @@
+/* Copyright (c) 2013-2022 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "LibraryModel.h"
+
+#include "../utils.h"
+
+#include <QApplication>
+#include <QDir>
+#include <QItemSelectionModel>
+#include <QSortFilterProxyModel>
+#include <QStyle>
+
+#include <algorithm>
+
+using namespace QGBA;
+
+LibraryModel::LibraryModel(QObject* parent)
+	: QAbstractItemModel(parent)
+	, m_treeMode(false)
+	, m_showFilename(false)
+{
+	QIcon iconGBA;
+	iconGBA.addFile(":/res/gba-icon-256.png", QSize(256, 256));
+	iconGBA.addFile(":/res/gba-icon-128.png");
+
+	QIcon iconGBC;
+	iconGBC.addFile(":/res/gbc-icon-128.png", QSize(128, 128));
+	iconGBC.addFile(":/res/gbc-icon-256.png", QSize(256, 256));
+
+	QIcon iconGB;
+	iconGB.addFile(":/res/gb-icon-128.png", QSize(128, 128));
+	iconGB.addFile(":/res/gb-icon-256.png", QSize(256, 256));
+
+	// QIcon iconNDS;
+	// iconNDS.addFile(":/res/gb-icon-128.png", QSize(128, 128));
+	// iconNDS.addFile(":/res/gb-icon-256.png", QSize(256, 256));
+
+	// These will silently and harmlessly fail if QSvgIconEngine isn't compiled in.
+	iconGBA.addFile(":/res/gba-icon.svg");
+	iconGBA.addFile(":/res/gbc-icon.svg");
+	iconGB.addFile(":/res/gb-icon.svg");
+	// iconNDS.addFile(":/res/nds-icon.svg");
+
+	m_icons[mPLATFORM_GBA] = iconGBA;
+	// m_icons[mPLATFORM_GBC] = iconGBC;
+	m_icons[mPLATFORM_GB] = iconGB;
+	// m_icons["NDS"] = iconNDS;
```

Comment:
`mPLATFORM_DS`

Reply:
Depends on the resolution of the linked PR.

Reply:
Went ahead and switched over to the data structure used by the GBC PR since it'll be more robust against future modifications.